src/jit/codegenxarch.cpp

```diff
@@ -1152,6 +1152,39 @@ void CodeGen::genReturn(GenTreePtr treeNode)
     // Also, there is not much to be gained by materializing it as an explicit node.
     if (compiler->compCurBB == compiler->genReturnBB)
     {
+#if defined(_TARGET_AMD64_)
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+        if (compiler->compMethodReturnsMultiRegRetType())
+        {
+            CORINFO_CLASS_HANDLE retClsHnd = compiler->info.compMethodInfo->args.retTypeClass;
+            ReturnTypeDesc retTypeDesc;
+            retTypeDesc.InitializeStructReturnType(compiler, retClsHnd);
+
+            unsigned regCount = retTypeDesc.GetReturnRegCount();
+
+            if (varTypeIsGC(compiler->info.compRetType))
```

Comment:
As I see the argument of  gcMarkRegSetNpt is regMaskTP not var_types, so I should replace 
```
gcInfo.gcMarkRegSetNpt(var_types);
```
by
```
gcInfo.gcMarkRegSetNpt(genRegMask(var_types));
```
Am I right?